src/sdks/core/src/cpp/sdk/cpptest/unit/advertisingTest.cpp

```diff
@@ -0,0 +1,99 @@
+#include "unit.h"
+
+
+class AdvertisingTest : public ::testing::Test {
+	protected:
+		JsonEngine* jsonEngine;
+		Firebolt::Error error = Firebolt::Error::None;
+
+	void SetUp() override
+	{
+		jsonEngine = new JsonEngine();
+	}
+
+	void TearDown() override
+	{
+		delete jsonEngine;
+	}
+
+	std::string skipRestrictionToString(Firebolt::Advertising::SkipRestriction skipRestriction) 
+	{
+		std::string str = "";
+		switch(skipRestriction) 
+		{
+			case Firebolt::Advertising::SkipRestriction::NONE: str = "none"; break;
+			case Firebolt::Advertising::SkipRestriction::ADS_UNWATCHED: str = "adsUnwatched"; break;
+			case Firebolt::Advertising::SkipRestriction::ADS_ALL: str = "adsAll"; break;
+			case Firebolt::Advertising::SkipRestriction::ALL: str = "all"; break;
+			default: str = "unknown";
+		}
+    	return str;
+	}
+
+};
+
+
+TEST_F(AdvertisingTest, Config)
+{
+	std::string expectedValues = jsonEngine->get_value("Advertising.config");
+
+	Firebolt::Advertising::AdConfigurationOptions options;
+	std::string adFrameworkConfig = Firebolt::IFireboltAccessor::Instance().AdvertisingInterface().config(options, &error);
+
+	EXPECT_EQ(error, Firebolt::Error::None);
+	EXPECT_EQ((nlohmann::json::parse(adFrameworkConfig)).dump(), expectedValues);
+}
+
+
+TEST_F(AdvertisingTest, Policy)
+{
+	nlohmann::json_abi_v3_11_3::json expectedValues = nlohmann::json::parse(jsonEngine->get_value("Advertising.policy"));
+
+	Firebolt::Advertising::AdPolicy adPolicy = Firebolt::IFireboltAccessor::Instance().AdvertisingInterface().policy(&error);
+
+	EXPECT_EQ(error, Firebolt::Error::None);
+
+	if(adPolicy.limitAdTracking.has_value())
+		EXPECT_EQ(adPolicy.limitAdTracking, expectedValues["limitAdTracking"]);
+	
+	if(adPolicy.skipRestriction.has_value())
+		EXPECT_EQ(skipRestrictionToString(adPolicy.skipRestriction.value()), expectedValues["skipRestriction"]);
+}
+
+
+TEST_F(AdvertisingTest, Id) {
+    nlohmann::json expectedValues = nlohmann::json::parse(jsonEngine->get_value("Advertising.advertisingId"));
+    
+    std::optional<Firebolt::Advertising::AdvertisingIdOptions> options = std::nullopt; // Assuming options are not provided
+
+    Firebolt::Advertising::AdvertisingId actualValues= Firebolt::IFireboltAccessor::Instance().AdvertisingInterface().advertisingId(options, &error);
+
+    ASSERT_EQ(error, Firebolt::Error::None) << "Error occurred during advertisingId call";
```

Comment:
1) Can we print the error we get if we get one?
2) Can this logging be done in all tests?

Reply:
Modified all tests with appropriate logging